crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs

```diff
@@ -748,93 +748,41 @@ pub(crate) fn manual_dict_lookup(
         return;
     }
 
-    // It's part of a bigger if-elif block:
-    // https://github.com/MartinThoma/flake8-simplify/issues/115
-    if let Some(Stmt::If(ast::StmtIf {
-        orelse: parent_orelse,
-        ..
-    })) = parent
-    {
-        if parent_orelse.len() == 1 && stmt == &parent_orelse[0] {
-            // TODO(charlie): These two cases have the same AST:
-            //
-            // if True:
-            //     pass
-            // elif a:
-            //     b = 1
-            // else:
-            //     b = 2
-            //
-            // if True:
-            //     pass
-            // else:
-            //     if a:
-            //         b = 1
-            //     else:
-            //         b = 2
-            //
-            // We want to flag the latter, but not the former. Right now, we flag neither.
-            return;
-        }
-    }
-
     let mut constants: FxHashSet<ComparableConstant> = FxHashSet::default();
     constants.insert(constant.into());
 
-    let mut child: Option<&Stmt> = orelse.get(0);
-    while let Some(current) = child.take() {
-        let Stmt::If(ast::StmtIf { test, body, orelse, range: _ }) = &current else {
-            return;
+    for clause in elif_else_clauses {
+        let ElifElseClause { test, body, .. } = clause;
+        let [Stmt::Return(ast::StmtReturn { value, range: _ })] = body.as_slice() else {
+            continue;
         };
-        if body.len() != 1 {
-            return;
-        }
-        if orelse.len() > 1 {
-            return;
-        }
-        let Expr::Compare(ast::ExprCompare {
+        if value.as_ref().map_or(false, |value| {
+            contains_effect(value, |id| checker.semantic().is_builtin(id))
+        }) {
+            continue;
+        };
```

Comment:
Nit: I don't know how expensive the `contains_effect` and `is_builtin` calls are but it could make sense to make this the very last check (testing if it is an elif or comparing the `id` and ops are relatively cheap)